### PR TITLE
Port over more internal components

### PR DESCRIPTION
**Pull request by stefanalund**
### StructuredMetadataTable
Screenshot 2020-04-13 at 14 53 03

### CopyTextButton
Screenshot 2020-04-13 at 17 43 03

### HorizontalProgress
Screenshot 2020-04-13 at 17 43 24

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/core/src/components/CopyTextButton/CopyTextButton.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  };
+
+  handleCopyClick = e => {
+    e.stopPropagation();
+    this.handleTooltipOpen();
+    try {
+      this.clipboardInput.select();
+      document.execCommand('copy');
+    } catch (error) {
+      // console.error(error);
```
Comment (stefanalund) [inline packages/core/src/components/CopyTextButton/CopyTextButton.js:77]: What should be done instead of logging? 🤔

**Comment by stefanalund** [comment]
@spotify/backstage-core ported some more components. Will create separate issues for converting these to TypeScript once this is merged.

File: packages/core/src/components/ProgressBars/HorizontalProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+type Props = {
+  /**
+   * Progress value between 0.0 - 1.0.
+   */
+  value: number;
+};
+
+const HorizontalProgress: FC<Props> = ({ value }) => {
+  if (isNaN(value)) {
```
Comment (stefanalund) [inline packages/core/src/components/ProgressBars/HorizontalProgress.tsx:31]: Is this check unnecessary when using TypeScript?

File: packages/core/src/components/CopyTextButton/CopyTextButton.js (hunk 1 above)
Comment (Rugvip) [reply packages/core/src/components/CopyTextButton/CopyTextButton.js:77]: Use the error api to report it to the app :grin:

```
const errorApi = useApi(errorApiRef);
errorApi.post(error);
```

Comment (stefanalund) [reply packages/core/src/components/CopyTextButton/CopyTextButton.js]: Fixed! Thanks

**Comment by stefanalund** [comment]
@Rugvip feel like pushing a change for your comments? I'm not sure I know what to do here 😆

**Review by stefanalund — COMMENTED** [review]
@Rugvip thanks for making the improvements. I guess we can merge this now. 

Will create a separate issue for converting JS -> TS.

**Review by Rugvip — APPROVED** [review]

**Pull request merged**